### ISSUE TITLE
Querying on joined tables

### ISSUE THREAD

**Issue by tory** [diogob/activerecord-postgres-earthdistance#23]
How does one go about using something like `within_radius` on a joined table?

Example:

``` ruby
class Job < ActiveRecord::Base
  belongs_to :event
end
```

``` ruby
class Event < ActiveRecord::Base
  has_many :jobs
  acts_as_geolocated latitude_column: 'lat', longitude_column: 'lng'
end
```

``` ruby
Job.includes(:event).where(filled: false).within_radius(100, -22.951916, -43.210487).references(:event)
```

This query expects the Job class to have a `latitude_column` and `longitude_column`. But I want to point to the `Event` table for that. I did not see a clear path to doing so. 

My solution has been to add an optional `klass` argument to the `within_radius` method, so that one can point to the joined table if desired, otherwise it falls back to the known class:

``` ruby
def within_radius(radius, lat, lng, klass=self) 
  earth_distance = Arel::Nodes::NamedFunction.new('earth_distance', [Utils.ll_to_earth_columns(klass), Utils.ll_to_earth_coords(lat, lng)])
  within_box(radius, lat, lng, klass).where(Arel::Nodes::InfixOperation.new('<=', earth_distance, Utils.quote_value(radius)))
end
```

If there's a better way to do this, I'd love to know. Otherwise I can flesh this idea out with a PR in the `ActsAsGeolocated::ClassMethods` module.

**Comment by tory** [comment]
bump, @diogob any thoughts on this?

**Comment by tory** [comment]
Resolved by #25 - thanks!
